addons/bitbucket/models.py

```diff
@@ -193,6 +193,11 @@ def is_private(self):
         connection = BitbucketClient(access_token=self.api.fetch_access_token())
         return connection.repo(user=self.user, repo=self.repo)['is_private']
 
+    @property
+    def ensure_repo(self):
```

Comment:
This property name doesn't reflect what you are using it for. You are fetching the repo, and if it is None, then you add a message that the repo has been deleted.  Maybe change to 'retrieved_repo', or 'fetched_repo', or turn this into a boolean and make it 'repo_exists'.    I thought that "ensure_repo" was doing other things based on the name.